src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/DesignTimeInputs.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    internal class DesignTimeInputs
+    {
+        public string[] Inputs = Array.Empty<string>();
+        public string[] SharedInputs = Array.Empty<string>();
```

Comment:
Consider making this type immutable. I've only seen immutable data passed around with Dataflow and assume mutable types can lead to corruption if a consumer starts modifying the data.

Similarly consider using `ImmutableArray<string>`.

Can `DesignTimeInputs` be a `readonly struct`?

Reply:
Yeah, as mentioned in the description the shape of this will change.